koku/masu/api/upgrade_trino/util/verify_parquet_files.py

```diff
@@ -0,0 +1,302 @@
+        return [None]
+
+    def _get_bill_dates(self):
+        # However far back we want to fix.
+        return [ciso8601.parse_datetime(self.bill_date)]
+
+    # Stolen from parquet_report_processor
+    def _parquet_path_s3(self, bill_date, report_type):
+        """The path in the S3 bucket where Parquet files are loaded."""
+        return get_path_prefix(
+            self.schema_name,
+            self.provider_type,
+            self.provider_uuid,
+            bill_date,
+            Config.PARQUET_DATA_TYPE,
+            report_type=report_type,
+        )
+
+    # Stolen from parquet_report_processor
+    def _parquet_daily_path_s3(self, bill_date, report_type):
+        """The path in the S3 bucket where Parquet files are loaded."""
+        if report_type is None:
+            report_type = "raw"
+        return get_path_prefix(
+            self.schema_name,
+            self.provider_type,
+            self.provider_uuid,
+            bill_date,
+            Config.PARQUET_DATA_TYPE,
+            report_type=report_type,
+            daily=True,
+        )
+
+    # Stolen from parquet_report_processor
+    def _parquet_ocp_on_cloud_path_s3(self, bill_date):
+        """The path in the S3 bucket where Parquet files are loaded."""
+        return get_path_prefix(
+            self.schema_name,
+            self.provider_type,
+            self.provider_uuid,
+            bill_date,
+            Config.PARQUET_DATA_TYPE,
+            report_type=OPENSHIFT_REPORT_TYPE,
+            daily=True,
+        )
+
+    # Stolen from parquet_report_processor
+    def _generate_s3_path_prefixes(self, bill_date):
+        """
+        generates the s3 path prefixes.
+        """
+        with schema_context(self.schema_name):
+            ocp_on_cloud_check = Provider.objects.filter(
+                infrastructure__infrastructure_provider__uuid=self.provider_uuid
+            ).exists()
+        path_prefixes = set()
+        for report_type in self.report_types:
+            path_prefixes.add(self._parquet_path_s3(bill_date, report_type))
+            path_prefixes.add(self._parquet_daily_path_s3(bill_date, report_type))
+            if ocp_on_cloud_check:
+                path_prefixes.add(self._parquet_ocp_on_cloud_path_s3(bill_date))
+        return path_prefixes
+
+    # Stolen from parquet_report_processor
+    @property
+    def local_path(self):
+        local_path = Path(Config.TMP_DIR, self.schema_name, str(self.provider_uuid))
+        local_path.mkdir(parents=True, exist_ok=True)
+        return local_path
+
+    def retrieve_verify_reload_s3_parquet(self):
+        """Retrieves the s3 files from s3"""
+        s3_resource = get_s3_resource(settings.S3_ACCESS_KEY, settings.S3_SECRET, settings.S3_REGION)
+        s3_bucket = s3_resource.Bucket(settings.S3_BUCKET_NAME)
+        bill_dates = self._get_bill_dates()
+        for bill_date in bill_dates:
```

Comment:
Right now we are only going to iterate over a single month based on the passed in `self.bill_date` see my comment above if we do want to run multiple months with a single trigger.